app/src/main/java/com/codesoom/assignment/application/ProductQueryService.java

```diff
@@ -0,0 +1,34 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.ProductNotFoundException;
+import com.codesoom.assignment.domain.Product;
+
+import java.util.Collection;
+
+/**
+ * 상품의 조회 기능을 제공합니다.
+ */
+public interface ProductQueryService {
+    /**
+     * 식별자를 가진 상품을 리턴한다.
+     *
+     * @param id 식별자
+     * @return 상품
+     * @throws ProductNotFoundException 식별자를 가진 상품이 없으면 예외를 던진다
+     */
+    Product findById(Long id);
```

Comment:
이렇게 한번 해보세요.
```suggestion
    /**
     * 상품을 찾아 리턴한다.
     *
     * @param id 식별자
     * @return 상품
     * @throws ProductNotFoundException 상품을 찾지 못한 경우
     */
    Product findById(Long id);
```

한편 `@throws`에 있는 `상품이 없으면`에 대해 곰곰히 생각해 보세요. 컴퓨터 시스템은 다양한 이유로 예외가 던져질 수 있습니다. 어떤 경우에는 DB에 상품이 있어도 정전이 되거나 네트워크 딜레이가 길어지거나 해서 있는 상품이 없다고 예외가 던져지기도 할 겁니다. 이런 경우를 잘 생각해보면 OS나 유명한 애플리케이션에서는 "찾지 못했습니다"라고 변명하는 상황을 본 적이 있을 겁니다. 절대로 "없다"라고 하지 않습니다. 이런 경우를 "겸손한 애플리케이션"이라고 부릅니다.

Reply:
큰 차이가 있네요!